### PR TITLE
Source fonts naming

### PR DESCRIPTION
**Pull request by heavywatal**
This PR addresses the issue #3867.
The following changes have been made.

- Source Sans 3: new cask
- Source Sans Pro: downgrade to version 2
- Source Serif 4: new cask
- Source Serif Pro: update file source and point to version 3

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).
- [x] `brew audit --cask {{cask_file}}` is error-free.
- [x] `brew style --fix {{cask_file}}` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask-fonts/search?q=is%3Aclosed&type=Issues).
- [x] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).
- [x] `brew audit --new-cask {{cask_file}}` worked successfully.
- [x] `brew install --cask {{cask_file}}` worked successfully.
- [x] `brew uninstall --cask {{cask_file}}` worked successfully.

### CODE REVIEW

**Comment by Loyalsoldier** [comment]
I think it's better to do like this:

Source Sans: new cask
Source Sans Pro: downgrade to version 2
Source Serif
Source Serif Pro: update file source and point to version 3

And use `caveats` to warn users about the newer release of casks like https://github.com/Homebrew/homebrew-cask-fonts/pull/3900

And deprecate `Source Sans Pro` and `Source Serif Pro` in a year or so.

**Comment by vitorgalvao** [comment]
What’s the point in keeping the old versions? They’re even failing `livecheck`, which should indicate the new ones are the continuation.

File: Casks/font-source-sans-3.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
+cask "font-source-sans-3" do
```
Comment (vitorgalvao) [inline Casks/font-source-sans-3.rb:1]: Versions shouldn’t be part of the name.

File: Casks/font-source-sans-3.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  version "3.028"
+  sha256 "1186d594ecb1336679a1d102bb6ddbe6041f0fd029482e53f02e46aa3ab011cf"
+
+  url "https://github.com/adobe-fonts/source-sans/releases/download/#{version}R/source-sans-#{version.sub(/\./, "v")}R.zip"
```
Comment (vitorgalvao) [inline Casks/font-source-sans-3.rb:5]:
```suggestion
  url "https://github.com/adobe-fonts/source-sans/releases/download/#{version}R/source-sans-#{version.major}v#{version.minor}R.zip"
```

File: Casks/font-source-sans-3.rb [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  name "Source Sans 3"
```
Comment (vitorgalvao) [inline Casks/font-source-sans-3.rb:6]: No version in name.

File: Casks/font-source-sans-3.rb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  desc "Fonts designed for user interfaces"
+  homepage "https://github.com/adobe-fonts/source-sans"
+
+  livecheck do
+    url "https://github.com/adobe-fonts/source-sans/releases/latest"
+    strategy :page_match
+    regex(%r{tag/(\d+(?:\.\d+)*)}i)
+  end
+
+  font "OTF/SourceSans3-Black.otf"
```
Comment (vitorgalvao) [inline Casks/font-source-sans-3.rb:16]:
```suggestion
  font "OTF/SourceSans#{version.major}-Black.otf"
```

In all of them.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by heavywatal** [comment]
Thank you for your feedback. I will consider `caveats` and the other requested changes after finishing this discussion.

But first, please read the upstream issues linked in #3867 again. "Source Sans 3" is not just the version 3 of "Source Sans", but it is the new family name. There is no font family named "Source Sans". Besides, the new one seems not geometrically upward-compatible with old "Source Sans Pro". In other words, silent upgrading from "Source Sans Pro" to "Source Sans 3" (and removing old "Source Sans Pro") will break existing documents. And it is expected to happen again when a new font family like "Source Sans 4" is released.

(I personally don't like their decision, but) keeping old fonts will be necessary to protect users' environments and documents.  Now I have two questions:

1. Based on the fact that "Source Sans 3" is the family name, should the cask be named `source-sans-3` or `source-sans`( and `source-sans@3` in the future)?
2. Is it possible to bypass `livecheck` for casks without `livecheck` entry?

**Comment by Loyalsoldier** [comment]
Is there any alias in this repo as the ones in `Homebrew/homebrew-core`?

Is so, the old `source sans pro` can then be linked to `source sans`, and `source serif pro` be linked to `source serif`.

**Comment by vitorgalvao** [comment]
> "Source Sans 3" is not just the version 3 of "Source Sans", but it is the new family name.

I mean, kind of. The repo doesn’t have the number in the name, nor does the first sentence in the README.

image

> (I personally don't like their decision, but)

Neither do I, but coming from Adobe I’m not surprised. We’ve removed plenty of their software from Homebrew Cask for user-hostility, and I’m inclined to do the same for these fonts.

[Our current preferred way to add fonts to this repo is via Google Fonts](https://github.com/Homebrew/homebrew-cask-fonts/blob/master/CONTRIBUTING.md#when-to-contribute), and I see they do have the Pro version. So it’s probably best to get rid of these and let Google Fonts deal with the naming however deem best; we’ll follow it automatically.

> 1. Based on the fact that "Source Sans 3" is the family name, should the cask be named `source-sans-3` or `source-sans`( and `source-sans@3` in the future)?

We don’t do the `@version` for casks and definitely don’t keep different versions of fonts. They already are more trouble than they’re worth without that.

> 2\. Is it possible to bypass `livecheck` for casks without `livecheck` entry?

CI always runs the check.

Sorry (and thank you) for the spent effort for a refusal, but it was necessary to see this in context to understand the situation. You can still use the fonts in your personal tap, though.

Will leave it open for a bit in case there are other ideas. Or perhaps they’ll come to their senses (though I doubt it).

**Comment by heavywatal** [comment]
OK, agree that they should be automatically managed via Google Fonts. And now I understand that each cask name should correspond to its distribution, not fonts in it. Thank you for reviewing the issue and PR.

FYI, by "family name" I meant the attribute of the font files (how it is referenced in an OS):
Screen Shot 2021-04-18 at 0 44 17

**Pull request closed without merging**